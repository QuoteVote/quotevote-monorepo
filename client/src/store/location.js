/**
 * Location Redux Store
 * Manages user's current location and permission state
 */

import { createSlice } from '@reduxjs/toolkit';
import { getCurrentPosition, checkPermission, GeoLocationError } from '../utils/geolocationService';

const initialState = {
  currentLocation: null, // { lat, lng }
  permissionState: 'prompt', // 'prompt' | 'granted' | 'denied' | 'unsupported'
  isLoading: false,
  error: null, // { type, message }
  lastUpdated: null,
};

const locationSlice = createSlice({
  name: 'location',
  initialState,
  reducers: {
    setLocation: (state, action) => {
      state.currentLocation = action.payload;
      state.lastUpdated = Date.now();
      state.error = null;
    },
    setPermissionState: (state, action) => {
      state.permissionState = action.payload;
    },
    setLoading: (state, action) => {
      state.isLoading = action.payload;
    },
    setError: (state, action) => {
      state.error = action.payload;
      state.isLoading = false;
    },
    clearError: (state) => {
      state.error = null;
    },
    clearLocation: (state) => {
      state.currentLocation = null;
      state.lastUpdated = null;
      state.error = null;
    },
    reset: () => initialState,
  },
});

export const {
  setLocation,
  setPermissionState,
  setLoading,
  setError,
  clearError,
  clearLocation,
  reset,
} = locationSlice.actions;

// Thunks

/**
 * Fetch user's current location
 * @param {object} options - Options passed to getCurrentPosition
 */
export const fetchLocation = (options = {}) => async (dispatch) => {
  dispatch(setLoading(true));
  dispatch(clearError());

  try {
    const location = await getCurrentPosition(options);
    dispatch(setLocation(location));
    dispatch(setPermissionState('granted'));
    return location;
  } catch (error) {
    dispatch(setError({
      type: error.type || GeoLocationError.UNKNOWN,
      message: error.message || 'Failed to get location',
    }));
    
    if (error.type === GeoLocationError.PERMISSION_DENIED) {
      dispatch(setPermissionState('denied'));
    }
    
    throw error;
  } finally {
    dispatch(setLoading(false));
  }
};

/**
 * Check and update permission state
 */
export const updatePermissionState = () => async (dispatch) => {
  try {
    const state = await checkPermission();
    dispatch(setPermissionState(state));
    return state;
  } catch (error) {
    console.error('Failed to check permission:', error);
    return 'prompt';
  }
};

// Selectors

export const selectLocation = (state) => state.location.currentLocation;
export const selectPermissionState = (state) => state.location.permissionState;
export const selectIsLoading = (state) => state.location.isLoading;
export const selectError = (state) => state.location.error;
export const selectLastUpdated = (state) => state.location.lastUpdated;

export const selectHasLocation = (state) => state.location.currentLocation !== null;
export const selectIsPermissionGranted = (state) => state.location.permissionState === 'granted';
export const selectIsPermissionDenied = (state) => state.location.permissionState === 'denied';

export default locationSlice.reducer;
