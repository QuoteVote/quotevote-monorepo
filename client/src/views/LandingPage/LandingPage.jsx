import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { tokenValidator } from 'store/user'
import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
import styles from 'assets/jss/material-dashboard-pro-react/views/landingPageStyle'
import {
  AppBar,
  Box,
  Card,
  Container,
  Divider,
  Grid,
  IconButton,
  Link,
  ListItem,
  Toolbar,
  Typography,
} from '@material-ui/core'
import { isMobile } from 'react-device-detect'
import GitHubIcon from '@material-ui/icons/GitHub'
import TwitterIcon from '@material-ui/icons/Twitter'
import LinkedInIcon from '@material-ui/icons/LinkedIn'
import EmailIcon from '@material-ui/icons/Email'
import GridContainer from '../../mui-pro/Grid/GridContainer'
import GridItem from '../../mui-pro/Grid/GridItem'
import Button from '../../mui-pro/CustomButtons/Button'

const useStyles = makeStyles((theme) => ({
  ...styles(theme),
  navbar: {
    background: 'linear-gradient(135deg, rgba(255,255,255,0.95) 0%, rgba(248,249,250,0.95) 100%)',
    backdropFilter: 'blur(10px)',
    boxShadow: '0 2px 8px rgba(0,0,0,0.1)',
    borderBottom: '2px solid transparent',
    borderImage: 'linear-gradient(90deg, #2AE6B2, #27C4E1, #178BE1) 1',
  },
  toolbar: {
    minHeight: 64,
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: theme.spacing(0, 3),
    [theme.breakpoints.down('sm')]: {
      minHeight: 56,
      padding: theme.spacing(0, 2),
      flexDirection: 'column',
      gap: theme.spacing(2),
    },
  },
  logo: {
    display: 'flex',
    alignItems: 'center',
    textDecoration: 'none',
    cursor: 'pointer',
    '&:hover': {
      opacity: 0.9,
    },
  },
  logoImage: {
    height: 40,
    width: 40,
  },
  brandText: {
    marginLeft: theme.spacing(1),
    fontWeight: 800,
    letterSpacing: '0.05em',
    color: '#0A2342',
    fontSize: '1.25rem',
  },
  navLinks: {
    display: 'flex',
    alignItems: 'center',
    gap: theme.spacing(2),
    [theme.breakpoints.down('sm')]: {
      flexDirection: 'column',
      width: '100%',
      gap: theme.spacing(1),
    },
  },
  navButton: {
    color: '#0A2342',
    fontWeight: 500,
    textTransform: 'none',
    padding: theme.spacing(1, 2),
    transition: 'all 0.2s',
    '&:hover': {
      background: 'rgba(42, 230, 178, 0.1)',
      transform: 'translateY(-1px)',
    },
    [theme.breakpoints.down('sm')]: {
      width: '100%',
      justifyContent: 'center',
    },
  },
  primaryButton: {
    background: 'linear-gradient(90deg, #2AE6B2, #27C4E1)',
    color: '#ffffff',
    fontWeight: 600,
    textTransform: 'none',
    padding: theme.spacing(1, 3),
    transition: 'all 0.2s',
    '&:hover': {
      background: 'linear-gradient(90deg, #27C4E1, #178BE1)',
      transform: 'translateY(-1px)',
      boxShadow: '0 4px 12px rgba(42, 230, 178, 0.3)',
    },
    [theme.breakpoints.down('sm')]: {
      width: '100%',
    },
  },
  outlinedButton: {
    border: '2px solid #2AE6B2',
    color: '#0A2342',
    fontWeight: 600,
    textTransform: 'none',
    padding: theme.spacing(1, 3),
    transition: 'all 0.2s',
    '&:hover': {
      background: 'rgba(42, 230, 178, 0.1)',
      transform: 'translateY(-1px)',
      border: '2px solid #2AE6B2',
    },
    [theme.breakpoints.down('sm')]: {
      width: '100%',
    },
  },
  footer: {
    background: 'linear-gradient(135deg, #0A2342 0%, #1a3a5c 100%)',
    color: '#ffffff',
    padding: theme.spacing(6, 0, 3),
    marginTop: theme.spacing(8),
  },
  footerContent: {
    maxWidth: 1200,
    margin: '0 auto',
    padding: theme.spacing(0, 3),
  },
  footerSection: {
    marginBottom: theme.spacing(4),
  },
  footerTitle: {
    fontWeight: 700,
    fontSize: '1.125rem',
    marginBottom: theme.spacing(2),
    color: '#2AE6B2',
  },
  footerLink: {
    color: '#ffffff',
    textDecoration: 'none',
    display: 'block',
    marginBottom: theme.spacing(1),
    transition: 'all 0.2s',
    '&:hover': {
      color: '#2AE6B2',
      transform: 'translateX(4px)',
    },
  },
  socialIcons: {
    display: 'flex',
    gap: theme.spacing(2),
    marginTop: theme.spacing(2),
  },
  socialIcon: {
    color: '#ffffff',
    transition: 'all 0.2s',
    '&:hover': {
      color: '#2AE6B2',
      transform: 'scale(1.1)',
    },
  },
  footerBottom: {
    borderTop: '1px solid rgba(255, 255, 255, 0.1)',
    marginTop: theme.spacing(4),
    paddingTop: theme.spacing(3),
    textAlign: 'center',
  },
  tagline: {
    fontStyle: 'italic',
    color: 'rgba(255, 255, 255, 0.8)',
    marginBottom: theme.spacing(2),
  },
}))

export const MOBILE_IMAGE_WIDTH = 250

export default function LandingPage() {
  const classes = useStyles({ isMobile })
  const dispatch = useDispatch()
  const history = useHistory()

  React.useEffect(() => {
    if (tokenValidator(dispatch)) history.push('/search')
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <Box style={{ minHeight: '100vh', display: 'flex', flexDirection: 'column' }}>
      {/* Enhanced Navbar */}
      <AppBar position="sticky" className={classes.navbar} elevation={0}>
        <Toolbar className={classes.toolbar}>
          {/* Logo */}
          <Box className={classes.logo} onClick={() => history.push('/')}>
            <img
              src="/assets/QuoteVoteLogo.png"
              alt="Quote.Vote Logo"
              className={classes.logoImage}
            />
            <Typography variant="h6" className={classes.brandText}>
              QUOTE.VOTE
            </Typography>
          </Box>

          {/* Navigation Links */}
          <Box className={classes.navLinks}>
            <Button
              className={classes.navButton}
              onClick={() => history.push('/')}
              aria-label="Home"
            >
              Home
            </Button>
            <Button
              className={classes.navButton}
              onClick={() => {
                const aboutSection = document.getElementById('about-section')
                aboutSection?.scrollIntoView({ behavior: 'smooth' })
              }}
              aria-label="About"
            >
              About
            </Button>
            <Button
              className={classes.navButton}
              href="mailto:admin@quote.vote"
              target="_blank"
              rel="noopener noreferrer"
              aria-label="Donate"
            >
              Donate
            </Button>
            <Button
              className={classes.outlinedButton}
              onClick={() => history.push('/auth/login')}
              aria-label="Login to your account"
            >
              Login
            </Button>
            <Button
              className={classes.primaryButton}
              onClick={() => history.push('/auth/request-access')}
              aria-label="Request an invite to join"
            >
              Request Invite
            </Button>
          </Box>
        </Toolbar>
      </AppBar>

      {/* Main Content */}
      <GridContainer
        direction="column"
        alignItems="center"
        justify="center"
        className={classes.container}
        style={{ flex: 1 }}
      >
        <GridItem xs={12} className={classes.listContainer}>
          <ListItem className={classes.listItem}>
            <img alt="Quote.Vote" src="/assets/QuoteVoteLogo.png" className={classes.voxPop} />
          </ListItem>

          <ListItem className={classes.listItem} id="about-section">
            <Card className={classes.card}>
              <Typography variant="h4" className={classes.listItemHeaderText}>
                Welcome to Quote.Vote
              </Typography>
              <Typography className={classes.listItemText}>
                Quote Vote is a non-for-profit project, and encourages users to donate their money or time, to be an active part of the change we&apos;d all like to see in the world.
              </Typography>
              <Typography className={classes.listItemText}>
                We understand the delicate balance between fostering freedom of expression and curbing harmful behavior.
              </Typography>
              <Typography className={classes.listItemText}>
                Our moderation policies aim to maximize the benefits of free speech while minimizing the potential for harm.
              </Typography>
              <Typography className={classes.listItemText}>
                We believe that thoughtful, respectful discourse leads to stronger communities and richer dialogue.
              </Typography>
            </Card>
          </ListItem>
        </GridItem>
      </GridContainer>

      {/* Enhanced Footer */}
      <Box component="footer" className={classes.footer} role="contentinfo">
        <Container className={classes.footerContent}>
          <Grid container spacing={4}>
            {/* About Section */}
            <Grid item xs={12} sm={6} md={3}>
              <Box className={classes.footerSection}>
                <Typography variant="h6" className={classes.footerTitle}>
                  About
                </Typography>
                <Typography variant="body2" className={classes.tagline}>
                  Empowering thoughtful discourse and community engagement through democratic quote voting.
                </Typography>
                <Link
                  href="mailto:admin@quote.vote"
                  className={classes.footerLink}
                  aria-label="Contact us via email"
                >
                  <EmailIcon style={{ fontSize: 16, verticalAlign: 'middle', marginRight: 8 }} />
                  admin@quote.vote
                </Link>
              </Box>
            </Grid>

            {/* Quick Links Section */}
            <Grid item xs={12} sm={6} md={3}>
              <Box className={classes.footerSection}>
                <Typography variant="h6" className={classes.footerTitle}>
                  Quick Links
                </Typography>
                <Link
                  href="/auth/request-access"
                  className={classes.footerLink}
                  aria-label="Request an invite"
                >
                  Request Invite
                </Link>
                <Link
                  href="/auth/login"
                  className={classes.footerLink}
                  aria-label="Login to your account"
                >
                  Login
                </Link>
                <Link
                  href="mailto:admin@quote.vote"
                  target="_blank"
                  rel="noopener noreferrer"
                  className={classes.footerLink}
                  aria-label="Donate to support us"
                >
                  Donate
                </Link>
                <Link
                  href="mailto:admin@quote.vote"
                  target="_blank"
                  rel="noopener noreferrer"
                  className={classes.footerLink}
                  aria-label="Volunteer with us"
                >
                  Volunteer
                </Link>
              </Box>
            </Grid>

            {/* Resources Section */}
            <Grid item xs={12} sm={6} md={3}>
              <Box className={classes.footerSection}>
                <Typography variant="h6" className={classes.footerTitle}>
                  Resources
                </Typography>
                <Link
                  href="/TERMS.md"
                  className={classes.footerLink}
                  aria-label="Read our terms of service"
                >
                  Terms of Service
                </Link>
                <Link
                  href="/quote_vote_code_of_conduct.md"
                  className={classes.footerLink}
                  aria-label="Read our code of conduct"
                >
                  Code of Conduct
                </Link>
                <Link
                  href="/CONTRIBUTING.md"
                  className={classes.footerLink}
                  aria-label="Learn how to contribute"
                >
                  Contributing
                </Link>
              </Box>
            </Grid>

            {/* Social Section */}
            <Grid item xs={12} sm={6} md={3}>
              <Box className={classes.footerSection}>
                <Typography variant="h6" className={classes.footerTitle}>
                  Connect With Us
                </Typography>
                <Box className={classes.socialIcons}>
                  <IconButton
                    href="https://github.com/QuoteVote/quotevote-monorepo"
                    target="_blank"
                    rel="noopener noreferrer"
                    className={classes.socialIcon}
                    aria-label="Visit our GitHub repository"
                  >
                    <GitHubIcon />
                  </IconButton>
                  <IconButton
                    href="https://twitter.com/quotevote"
                    target="_blank"
                    rel="noopener noreferrer"
                    className={classes.socialIcon}
                    aria-label="Follow us on Twitter"
                  >
                    <TwitterIcon />
                  </IconButton>
                  <IconButton
                    href="https://linkedin.com/company/quotevote"
                    target="_blank"
                    rel="noopener noreferrer"
                    className={classes.socialIcon}
                    aria-label="Connect with us on LinkedIn"
                  >
                    <LinkedInIcon />
                  </IconButton>
                </Box>
              </Box>
            </Grid>
          </Grid>

          {/* Footer Bottom */}
          <Box className={classes.footerBottom}>
            <Typography variant="body2" style={{ marginBottom: 8 }}>
              Made with <span style={{ color: '#e25555' }}>❤️</span> on Earth
            </Typography>
            <Typography variant="body2" style={{ opacity: 0.8 }}>
              © {new Date().getFullYear()} Quote.Vote. All rights reserved.
            </Typography>
          </Box>
        </Container>
      </Box>
    </Box>
  )
}
