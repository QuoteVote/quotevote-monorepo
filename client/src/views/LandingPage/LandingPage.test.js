import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import LandingPage from './LandingPage'
import store from '../../store/store'

// Mock the useHistory hook
const mockHistoryPush = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

const LandingPageWrapper = () => (
  <Provider store={store}>
    <BrowserRouter>
      <LandingPage />
    </BrowserRouter>
  </Provider>
)

describe('LandingPage test -', () => {
  beforeEach(() => {
    mockHistoryPush.mockClear()
  })

  it('renders correctly', () => {
    const { container } = render(<LandingPageWrapper />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders enhanced navbar with all navigation links', () => {
    render(<LandingPageWrapper />)
    
    // Check for navbar elements
    expect(screen.getByText('QUOTE.VOTE')).toBeInTheDocument()
    expect(screen.getByText('Home')).toBeInTheDocument()
    expect(screen.getByText('About')).toBeInTheDocument()
    expect(screen.getByText('Donate')).toBeInTheDocument()
    expect(screen.getByText('Login')).toBeInTheDocument()
    expect(screen.getByText('Request Invite')).toBeInTheDocument()
  })

  it('renders comprehensive footer with all sections', () => {
    render(<LandingPageWrapper />)
    
    // Check for footer elements
    expect(screen.getByText('About')).toBeInTheDocument()
    expect(screen.getByText('Quick Links')).toBeInTheDocument()
    expect(screen.getByText('Resources')).toBeInTheDocument()
    expect(screen.getByText('Connect With Us')).toBeInTheDocument()
    
    // Check for specific footer links
    expect(screen.getByText('Request Invite')).toBeInTheDocument()
    expect(screen.getByText('Login')).toBeInTheDocument()
    expect(screen.getByText('Donate')).toBeInTheDocument()
    expect(screen.getByText('Volunteer')).toBeInTheDocument()
    expect(screen.getByText('Terms of Service')).toBeInTheDocument()
    expect(screen.getByText('Code of Conduct')).toBeInTheDocument()
    expect(screen.getByText('Contributing')).toBeInTheDocument()
    
    // Check for copyright
    expect(screen.getByText(/Â© \d{4} Quote.Vote. All rights reserved./)).toBeInTheDocument()
  })

  it('handles navigation button clicks correctly', () => {
    render(<LandingPageWrapper />)
    
    // Test Home button
    fireEvent.click(screen.getByText('Home'))
    expect(mockHistoryPush).toHaveBeenCalledWith('/')
    
    // Test Login button
    fireEvent.click(screen.getByText('Login'))
    expect(mockHistoryPush).toHaveBeenCalledWith('/auth/login')
    
    // Test Request Invite button
    fireEvent.click(screen.getByText('Request Invite'))
    expect(mockHistoryPush).toHaveBeenCalledWith('/auth/request-access')
  })

  it('has proper accessibility attributes', () => {
    render(<LandingPageWrapper />)
    
    // Check for aria-labels on interactive elements
    expect(screen.getByLabelText('Home')).toBeInTheDocument()
    expect(screen.getByLabelText('About')).toBeInTheDocument()
    expect(screen.getByLabelText('Donate')).toBeInTheDocument()
    expect(screen.getByLabelText('Login to your account')).toBeInTheDocument()
    expect(screen.getByLabelText('Request an invite to join')).toBeInTheDocument()
    
    // Check for role attributes
    const footer = screen.getByRole('contentinfo')
    expect(footer).toBeInTheDocument()
  })
})
