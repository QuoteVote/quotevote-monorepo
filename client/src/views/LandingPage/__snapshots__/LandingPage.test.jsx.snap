// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LandingPage test - > renders correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-grid-50 makeStyles-container-2 css-1bkjq21-MuiGrid-root"
  justify="center"
>
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row makeStyles-grid-51 makeStyles-listContainer-8 css-17fpwt7-MuiGrid-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-listItem-9 css-cfd5wn-MuiListItem-root"
    >
      <img
        alt="quoteVote"
        class="makeStyles-voxPop-7"
        src="/assets/QuoteVoteLogo.png"
      />
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-listItem-9 css-cfd5wn-MuiListItem-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary makeStyles-button-52 makeStyles-listItemTextLogin-11 css-dx1o0x-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Login
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary makeStyles-button-52 makeStyles-listItemTextRequestInvite-10 css-59cs2h-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Request Invite
      </button>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-listItem-9 css-cfd5wn-MuiListItem-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-12 css-1j74uew-MuiPaper-root-MuiCard-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 makeStyles-listItemHeaderText-13 css-f84ov1-MuiTypography-root"
        >
          Welcome to Quote.Vote
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body1 makeStyles-listItemText-14 css-rizt0-MuiTypography-root"
        >
          Quote Vote is a non-for-profit project, and encourages users to donate their money or time, to be an active part of the change we'd all like to see in the world.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 makeStyles-listItemText-14 css-rizt0-MuiTypography-root"
        >
          We understand the delicate balance between fostering freedom of expression and curbing harmful behavior.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 makeStyles-listItemText-14 css-rizt0-MuiTypography-root"
        >
          Our moderation policies aim to maximize the benefits of free speech while minimizing the potential for harm.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 makeStyles-listItemText-14 css-rizt0-MuiTypography-root"
        >
          We believe that thoughtful, respectful discourse leads to stronger communities and richer dialogue.
        </p>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-listItem-9 css-cfd5wn-MuiListItem-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary makeStyles-button-52 makeStyles-listItemTextDonate-15 css-59cs2h-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Donate
      </button>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-contact-16 css-cfd5wn-MuiListItem-root"
    >
      Contact
       
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-q3dtp5-MuiTypography-root-MuiLink-root"
        href="mailto:admin@quote.vote"
        rel="noreferrer"
        target="_black"
      >
        admin@quote.vote
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding makeStyles-listItem-9 css-cfd5wn-MuiListItem-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 makeStyles-listItemMakeWithLove-17 css-rizt0-MuiTypography-root"
      >
        Made with
         
        <span>
          â™¥
        </span>
         
        on Earth
      </p>
    </li>
  </div>
</div>
`;
